instrumentation/grpc-1.5/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/grpc/v1_5/GrpcServerBuilderInstrumentation.java

```diff
@@ -21,33 +28,61 @@
 
 public class GrpcServerBuilderInstrumentation implements TypeInstrumentation {
 
+  @Override
+  public ElementMatcher<ClassLoader> classLoaderOptimization() {
+    return hasClassesNamed("io.grpc.ServerBuilder");
+  }
+
   @Override
   public ElementMatcher<TypeDescription> typeMatcher() {
-    return named("io.grpc.internal.AbstractServerImplBuilder");
+    return safeHasSuperType(named("io.grpc.ServerBuilder"));
   }
 
   @Override
   public Map<? extends ElementMatcher<? super MethodDescription>, String> transformers() {
-    return singletonMap(
-        isMethod().and(named("build")),
-        GrpcServerBuilderInstrumentation.class.getName() + "$AddInterceptorAdvice");
+    Map<ElementMatcher<MethodDescription>, String> transformers = new HashMap<>();
+    transformers.put(
+        isMethod()
+            .and(isPublic())
+            .and(named("intercept"))
+            .and(takesArgument(0, named("io.grpc.ServerInterceptor"))),
+        GrpcServerBuilderInstrumentation.class.getName() + "$InterceptAdvice");
+    transformers.put(
+        isMethod().and(isPublic()).and(named("build")).and(takesArguments(0)),
+        GrpcServerBuilderInstrumentation.class.getName() + "$BuildAdvice");
+    return transformers;
   }
 
-  public static class AddInterceptorAdvice {
+  public static class InterceptAdvice {
 
     @Advice.OnMethodEnter(suppress = Throwable.class)
-    public static void addInterceptor(
-        @Advice.FieldValue("interceptors") List<ServerInterceptor> interceptors) {
-      boolean shouldRegister = true;
-      for (ServerInterceptor interceptor : interceptors) {
-        if (interceptor instanceof TracingServerInterceptor) {
-          shouldRegister = false;
-          break;
-        }
+    public static void onEnter(
+        @Advice.This ServerBuilder<?> serverBuilder,
+        @Advice.Argument(0) ServerInterceptor interceptor) {
+      // Check against unshaded name.
+      if (interceptor
+          .getClass()
+          .getName()
+          .equals("io.opentelemetry.instrumentation.grpc.v1_5.server.TracingServerInterceptor")) {
```

Comment:
this string will get shaded too 😄

Reply:
Oops - removed the check for now so we can come up with a better strategy globally

Reply:
So, will we have multiple copies of `TracingServerInterceptor`?

Reply:
See #903